src/library/Avatar/Avatar.js

```diff
@@ -28,14 +27,16 @@ export default class Avatar extends Component<AvatarProps> {
           ) : (
             <span>{child}</span>
           );
-      } else if (
-        child.type &&
-        child.type.displayName &&
-        child.type.displayName.indexOf('Icon') !== -1
-      ) {
-        icon = cloneElement(child, {
-          size: ICON_SIZE[size || Avatar.defaultProps.size]
-        });
+      } else if (child.type) {
+        const componentName = child.type.displayName || child.type.name || '';
```

Comment:
Might be worth, creating a simple helper at the bottom of the file:

```ts
function getComponentName(element) {
  return element.type.displayName || element.type.name || '';
}
```
that would allow you to do:
```
} else if (child.type && getComponentName(child).indexOf('Icon') !== -1) {
...
) else {
```
without needing the nested `if`.

diff would be super minimal and clean with that change.